packages/lit-html/src/directives/choose.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Chooses and evaluates a template function from a list based on matching
+ * the given `value` to a case.
+ *
+ * Cases are structured as `[caseValue, func]`. `value` is matched to
+ * `caseValue` by strict equality. The first match is selected.
+ *
+ * This is similar to a switch statement, but as an expression and without
+ * fallthrough.
+ */
+export const choose = <T, V>(
+  value: T,
+  cases: Array<[T, () => V]>,
+  defaultCase?: () => V
+) => {
+  for (const c of cases) {
```

Comment:
For string cases, this could be an object index. But I assume it's an array search because we want to support more complex keys?

Reply:
yeah... and we're matching Wit's implementation for now. We could add an overload for objects for a nicer syntax syntax for string keys.